app/javascript/components/charts/components/chart-tooltip/chart-tooltip-component.js

```diff
@@ -23,7 +23,10 @@ class ChartTooltip extends PureComponent {
                 : values[d.key];
 
               return hideZeros && (!values || !value) ? null : (
-                <div key={d.key} className={`data-line ${d.position || ''}`}>
+                <div
+                  key={d.key ? d.key : `setting-${i}`}
```

Comment:
no use of i in keys please. React gets very upset about this.

Reply:
This solves an issue where the key is undefined and react complains about it. Not sure what to add then 🙃

Reply:
If we don't have a key we shouldn't be rendering it.

Reply:
If there is no unique data to identify what it is then what is it doing there? Check the data and the render to tidy it up.